script/subtitles/imscsubtitles.js

```diff
@@ -25,6 +26,22 @@ define('bigscreenplayer/subtitles/imscsubtitles',
         Plugins.interface.onSubtitlesTransformError();
       }
 
+      // Opts: { backgroundColour: string (css colour, hex), fontFamily: string }
+      function transformStyleOptions (opts) {
+        if (opts === undefined) return;
+
+        var customStyles = {};
+
+        if (opts.backgroundColour) {
+          customStyles.spanBackgroundColorAdjust = {transparent: opts.backgroundColour, black: opts.backgroundColour};
```

Comment:
Why also transform already black backgrounds? If you omit that the result should be the same.

Reply:
this implementation doesn't have knowledge if the colour we are transforming to, as it's passed in. hypothetically you might want to set it as another colour

Reply:
right, but then always transforming _black_ won't work since the implementation also doesn't know the colour it is transforming _from_. There are some old subs files around that do use "interesting" background colours.

Reply:
This may be an argument for a customisation that can map "any", which is something I've been wondering about already.

Then with a mapping table:
```json
{
   "*": "black",
   "black": "navy"
}
```

where `*` is some wildcard that matches everything,

should a black background passed in end up black or navy?